src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/SecurityRatingCalculator.java

```diff
@@ -192,6 +208,89 @@ private static void processConfig(String filename, GitHub github, String githubT
       System.out.print(new PrettyPrinter().print(ratingValue));
       System.out.println("[+]");
     }
+
+    storeProjectCache(projectCache, projectCacheFile);
+
+    for (Reporter<GitHubProject> reporter : reporters) {
+      reporter.runFor(projects);
+    }
+  }
+
+  /**
+   * Stores a cache of projects to a file.
+   *
+   * @param projectCache The cache of projects.
+   * @param projectCacheFile A path to the file.
+   * @throws IOException If something went wrong.
+   */
+  private static void storeProjectCache(GitHubProjectCache projectCache, String projectCacheFile)
+      throws IOException {
+
+    System.out.printf("[+] Storing the project cache to %s%n", projectCacheFile);
+    ObjectMapper mapper = new ObjectMapper();
+    Files.write(
+        Paths.get(projectCacheFile),
+        mapper.writerWithDefaultPrettyPrinter().writeValueAsBytes(projectCache));
+  }
+
+  /**
+   * Loads a cache of projects from a file.
+   * If the file doesn't exist, then the method returns an empty cache.
+   *
+   * @param filename A path to the file.
+   * @return A loaded cache of projects.
+   * @throws IOException If something went wrong.
+   */
+  private static GitHubProjectCache loadProjectCache(String filename) throws IOException {
+    if (Files.exists(Paths.get(filename))) {
+      System.out.printf("[+] Loading a project cache from %s%n", filename);
+      return GitHubProjectCache.load(filename);
+    }
+
+    return GitHubProjectCache.empty();
+  }
+
+  /**
+   * Returns a filename of a project cache.
+   *
+   * @param config A config of the tool.
+   * @return A file from the config if available, the default cache file otherwise.
+   */
+  private static String projectCacheFile(Config config) {
+    return config.hasCacheFile() ? config.cacheFilename : DEFAULT_PROJECT_CACHE_FILE;
+  }
+
+  /**
+   * Create a reporter from a config.
+   * The method returns {@link Reporter#DUMMY} if the config is null.
+   *
+   * @param config The config.
+   * @return A reporter.
+   * @throws IllegalArgumentException If the type is unknown.
+   */
+  private static List<Reporter<GitHubProject>> makeReporters(Config config) throws IOException {
+    Objects.requireNonNull(config, "Oh no! Config is null!");
+    if (config.reportConfigs == null) {
+      return Collections.singletonList(Reporter.DUMMY);
+    }
+
+    List<Reporter<GitHubProject>> reporters = new ArrayList<>();
+    for (ReportConfig reportConfig : config.reportConfigs) {
+      Objects.requireNonNull(reportConfig.type, "Hey! Reporter type can't be null!");
+      switch (reportConfig.type) {
+        case "markdown":
```

Comment:
possible to extract these "markdown" , "json" to constants.

Reply:
Yup, I thought about introducing a enum. I think it would be even better.

Reply:
Yes.